src/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -0,0 +1,193 @@
+import org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerException;
+import org.zaproxy.zap.extension.openapi.network.Requestor;
+import org.zaproxy.zap.extension.spider.ExtensionSpider;
+import org.zaproxy.zap.spider.parser.SpiderParser;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionOpenApi extends ExtensionAdaptor {
+
+    public static final String NAME = "ExtensionOpenApi";
+
+    private static final String THREAD_PREFIX = "ZAP-Import-OpenAPI-";
+
+    private ZapMenuItem menuImportLocalOpenApi = null;
+    private ZapMenuItem menuImportUrlOpenApi = null;
+    private int threadId = 1;
+
+    private static final Logger LOG = Logger.getLogger(ExtensionOpenApi.class);
+
+    public ExtensionOpenApi() {
+        super(NAME);
+        this.setOrder(158);
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        if (getView() != null) {
+            extensionHook.getHookMenu().addToolsMenuItem(getMenuImportLocalOpenApi());
+            extensionHook.getHookMenu().addToolsMenuItem(getMenuImportUrlOpenApi());
+
+            /* Custom spider is added in order to explore Open API definitions. */
+            OpenApiSpider.enable();
+            ExtensionSpider spider = (ExtensionSpider) Control.getSingleton()
+                    .getExtensionLoader()
+                    .getExtension(ExtensionSpider.NAME);
+            SpiderParser customSpider = new OpenApiSpider();
+            if (spider != null) {
+                spider.addCustomParser(customSpider);
+                LOG.info("Added custom Open API spider.");
+            } else {
+                LOG.info("Custom Open API spider could not be added.");
+            }
+        }
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        /* Disables custom spider. */
+        OpenApiSpider.disable();
+    }
+
+    /* Menu option to import a local OpenApi file. */
+    private ZapMenuItem getMenuImportLocalOpenApi() {
+        if (menuImportLocalOpenApi == null) {
+            menuImportLocalOpenApi = new ZapMenuItem("openapi.topmenu.tools.importOpenApi");
+            menuImportLocalOpenApi.setToolTipText(Constant.messages.getString("openapi.topmenu.tools.importOpenApi.tooltip"));
+
+            menuImportLocalOpenApi.addActionListener(new java.awt.event.ActionListener() {
+
+                @Override
+                public void actionPerformed(java.awt.event.ActionEvent e) {
+                    // Prompt for a OpenApi file.
+                    final JFileChooser chooser = new JFileChooser(Model.getSingleton().getOptionsParam().getUserDirectory());
+                    int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+                    if (rc == JFileChooser.APPROVE_OPTION) {
+                        extUrlOpenApiImport("file://" + chooser.getSelectedFile().getAbsolutePath());
```

Comment:
Oh, this _was_ working for me :/ did you try with a valid filename, or was that one valid?